File: src/algorithms/ValidAnagram.cpp
// define index size
const int MAX_INDEX_SIZE = 26;

class Solution {
public:
    bool isAnagram(string s, string t) {
        int count[MAX_INDEX_SIZE];

        memset(count, 0, sizeof(count));
        for (int i = 0; i < s.length(); i++)
            count[s[i] - 'a']++;
        for (int i = 0; i < t.length(); i++)
            count[t[i] - 'a']--;
        for (int i = 0; i < MAX_INDEX_SIZE; i++)
            if (count[i]) return false;
        return true;
    }
};

File: Algorithms/python/242-ValidAnagram/ValidAnagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        chars_count = dict()

        for c in s:
            if c not in chars_count:
                chars_count[c] = 1
            else:
                chars_count[c] += 1

        for c in t:
            if c not in chars_count:
                return False
            chars_count[c] -= 1

        for k in chars_count.keys():
            if chars_count[k] != 0:
                return False
        return True
